src/managing-workflow/security-considerations.md

```diff
@@ -1,3 +1,56 @@
 # Security Considerations
 
-TODO (bacongobbler): rewrite for v2
+!!! important
+	Workflow is not suitable for multi-tenant environments or hosting untrusted code.
+
+A major goal of Workflow is to be operationally secure and trusted by operations engineers in every
+deployed environment. There are, however, two notable security-related considerations to be aware
+of when deploying Workflow.
+
+# Application Runtime Segregation
+
+Users of Workflow often want to deploy their applications to separate environments (commonly:
+development, staging, and production). Typically, physical network isolation isn’t the goal, but
+rather segregation of application environments - if a development app goes haywire, it shouldn’t
+affect production applications that are running in the cluster.
+
+In Workflow, deployed applications can be segregated by using the `deis tags` command. This enables
+you to tag machines in your cluster with arbitrary metadata, then configure your applications to be
+scheduled to machines which match the metadata.
+
+For example, if some machines in your cluster are tagged with `environment=production` and some
+with `environment=staging`, you can configure an application to be deployed to the production
+environment by using `deis tags set environment=production`. Workflow will pass this configuration
+along to Kubernetes, which will schedule your application in different environments running on
+separate hardware.
+
```

Comment:
This would seem to indicate that we regard a tagged subset of worker nodes as an "environment" and I think that might be a bit of a stretch. If it's still a single Kubernetes cluster, simply segregating applications to a subset of nodes doesn't really achieve very much. I'll elaborate on this in a general comment below.